pkg/virtual/workspaces/registry/rest.go

```diff
@@ -163,27 +150,29 @@ func (s *REST) NamespaceScoped() bool {
 	return false
 }
 
-func (s *REST) getPrettyNameFromInternalName(user kuser.Info, internalName string) (string, error) {
+func (s *REST) getPrettyNameFromInternalName(user kuser.Info, orgName, internalName string) (string, error) {
 	list, err := s.crbInformer.Informer().GetIndexer().ByIndex(InternalNameIndex, internalName)
 	if err != nil {
 		return "", err
 	}
 	for _, el := range list {
 		if crb, isCRB := el.(*rbacv1.ClusterRoleBinding); isCRB &&
+			crb.ClusterName == orgName &&
```

Comment:
can't we look up by clusterName? E.g. have an OrgInternalNameIndex with `#$#` separator or similar.

Reply:
Yes, I thought about doing it, but as a first step started with the simplest and quickest option.